include/seqan3/alignment/pairwise/detail/pairwise_alignment_algorithm.hpp

```diff
@@ -165,54 +165,137 @@ class pairwise_alignment_algorithm : protected policies_t...
-        *alignment_column.begin() = this->initialise_origin_cell();
-        for (auto cell : alignment_column | seqan3::views::drop(1))
-            cell = this->initialise_first_column_cell(cell);
+        for ([[maybe_unused]] auto && unused : sequence2)
+            this->track_last_column_cell(*++column_iterator);
 
+        this->track_final_cell(*column_iterator);
+
+        return this->tracked_optimum();
+    }
+
+    /*!\brief Initialise the first column of the alignment matrix.
+     * \tparam alignment_matrix_column_t The type of the alignment matrix; must model std::ranges::input_range.
+     * \tparam sequence2_t The type of the second sequence; must model std::ranges::input_range.
+     *
+     * \param[in] first_column The first column of the alignment matrix to initialise.
+     * \param[in] sequence2 The second sequence used to determine the size of the column.
+     *
+     * \details
+     *
+     * The first column of the alignment matrix does not require any character comparisons of the sequences that
+     * shall be aligned. The second sequence is thus only needed to determine the size of the column.
+     * The computation of the column is split into three phases: the initialisation phase, the iteration phase, and
+     * the final phase. In the initialisation phase the first cell of the column is computed and in the iteration
+     * phase all remaining cells are computed. In the final phase the last cell is possibly evaluated for a new
+     * alignment optimum.
+     */
+    template <std::ranges::input_range alignment_matrix_column_t, std::ranges::input_range sequence2_t>
+    void initialise_column(alignment_matrix_column_t && first_column, sequence2_t && sequence2)
+    {
         // ---------------------------------------------------------------------
-        // Recursion phase: compute column-wise the alignment matrix.
+        // Initial phase: prepare column and initialise first cell
         // ---------------------------------------------------------------------
 
-        // Compute remaining matrix
-        ++matrix_iter; // Move to next column.
-        for (auto sequence1_iter = sequence1.begin();
-             sequence1_iter != sequence1.end();
-             ++sequence1_iter, ++matrix_iter)
+        auto first_column_iter = first_column.begin();
+        *first_column_iter = this->track_cell(this->initialise_origin_cell());
+
+        // ---------------------------------------------------------------------
+        // Iteration phase: iterate over column and compute each cell
+        // ---------------------------------------------------------------------
+
+        for ([[maybe_unused]] auto && unused : sequence2)
         {
-            alignment_column = *matrix_iter;
-            auto alignment_column_iter = alignment_column.begin();
+            ++first_column_iter;
+            *first_column_iter = this->track_cell(this->initialise_first_column_cell(*first_column_iter));
+        }
```

Comment:
This looks like the column is already resized to the size of sequence2, since you do not push_back but assign. Can you not query the size of an alignment column, then you would not need sequence2 for the size?

Reply:
I might not always have a sized range. In general, this is not a generic part since we control the matrix implementation but with combining them with views and modifying them it can happen that a column is not sized for any particular reason. 
I also don't want to go to the end of the column since I want to do something special with the last cell of the column. Since a column expects at most an input range (actually also an output range but that will come later), I cannot simply set the iterator to the one before the end. So in either case I have to introduce some additional variable to iterate over the column. A valid point would be to just iterate over the sequence and not dereference the value. But I couldn't find any implications when looking at the assembly. So far the results indicate no change. In the banded case I might safe some head-ach since the sequence slice will determine the actual size of the column.